packages/react/src/Form/core/useOptionsWithId.ts

```diff
@@ -0,0 +1,13 @@
+/* eslint-disable react-hooks/rules-of-hooks */
+import { useId } from "react";
+import type { Option } from ".";
+
+export function useOptionsWithId(options?: Option[], id?: string) {
+  if (!options) {
+    return [];
+  }
+  return options.map((option, index) => {
+    const newOptionId = id ? `${id}_${index}` : useId();
+    return { ...option, id: option.id ?? newOptionId };
+  });
+}
```

Comment:
Alors ici je ne suis pas fan de désactivait la règle `react-hooks/rules-of-hooks` pour tous le fichier
Et oui il ne faut pas conditionner l'utilisation des hook donc ici il faut faire:
```suggestion
import { useId } from "react";
import type { Option } from ".";

export function useOptionsWithId(options?: Option[], id?: string) {
  const reactId = useId();
  if (!options) {
    return [];
  }
  return options.map((option, index) => {
    const newOptionId = id ? `${id}_${index}` : reactId;
    return { ...option, id: option.id ?? newOptionId };
  });
}
```

Reply:
done